numba_rvsdg/core/transformations.py

```diff
@@ -151,7 +151,7 @@ def reverse_lookup(d, value):
                     # the correct blocks
                     variable_assignment = {}
                     variable_assignment[backedge_variable] = reverse_lookup(backedge_value_table, loop_head)
-                    if needs_synth_exit:
+                    if needs_synth_exit or headers_were_unified:
```

Comment:
So what this basically means:

If there's no need for a synthetic exit and the headers were unified in a loop restructure, then the variable assignment for exit variable (i.e. the exit variable would be assigned the value) equal to the one that points it to the jump target i.e. the newly created/unified header.

Is the above statement true?